cmd/finch/version.go

```
@@ -4,27 +4,107 @@
+type versionAction struct {
+	creator command.LimaCmdCreator
+	logger  flog.Logger
+	stdOut  io.Writer
+}
+
+// NerdctlVersionOutput captures the nerdctl version.
+type NerdctlVersionOutput struct {
+	Client NerdctlClientOuput  `json:"Client"`
+	Server NerdctlServerOutput `json:"Server"`
+}
+
+// NerdctlClientOuput captures the nerdctl Client output.
+type NerdctlClientOuput struct {
+	Version    string                    `json:"Version"`
+	GitCommit  string                    `json:"GitCommit"`
+	GoVersion  string                    `json:"GoVersion"`
+	Os         string                    `json:"Os"`
+	Arch       string                    `json:"Arch"`
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlServerOutput captures the nerdctl Server output.
+type NerdctlServerOutput struct {
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlComponentsOutput captures the nerdctl components output.
+type NerdctlComponentsOutput struct {
+	Name    string `json:"Name"`
+	Version string `json:"Version"`
+	Details struct {
+		GitCommit string `json:"GitCommit"`
+	}
+}
+
+func newVersionAction(creator command.LimaCmdCreator, logger flog.Logger, stdOut io.Writer) *versionAction {
+	return &versionAction{creator: creator, logger: logger, stdOut: stdOut}
+}
+
+func (va *versionAction) runAdapter(cmd *cobra.Command, args []string) error {
+	return va.run()
+}
+
+func (va *versionAction) run() error {
+	status, err := lima.GetVMStatus(va.creator, va.logger, limaInstanceName)
+	if (status != lima.Running) || (err != nil) {
+		fmt.Println("Finch version:", version.Version)
+		fmt.Println("Please run finch vm start to start the virtual machine")
+	} else {
+		limaArgs := []string{"shell", limaInstanceName, "nerdctl", "version", "--format", "json"}
+		out, _ := va.creator.CreateWithoutStdio(limaArgs...).Output()
+
+		var nerdctlVersion NerdctlVersionOutput
+		err = json.Unmarshal(out, &nerdctlVersion)
+
+		if err == nil {
```

if the error is not nil, we may want to return it instead of a silent fail.

---

updated